### PR TITLE
Migrate to DataStore

### PR DESCRIPTION
**Pull request by rock3r**
Thanks @florina-muntenescu for the help, here's your PR to double check :)

### CODE REVIEW

File: app/src/main/java/dev/sebastiano/bundel/preferences/DataStorePreferenceStorage.kt [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                        Timber.e(throwable)
+                        emit(true)
+                    }
+                    else -> {
+                        throw throwable
+                    }
+                }
+            }
+
+    override suspend fun setIsCrashlyticsEnabled(enabled: Boolean): Boolean =
```
Comment (florina-muntenescu) [inline app/src/main/java/dev/sebastiano/bundel/preferences/DataStorePreferenceStorage.kt:36]: Checking the code, it looks like the return type is never used.
How about dropping it for all `set...` calls?

File: app/src/main/java/dev/sebastiano/bundel/preferences/PreferenceStorage.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package dev.sebastiano.bundel.preferences
+
+import dev.sebastiano.bundel.preferences.schedule.TimeRangesSchedule
+import dev.sebastiano.bundel.preferences.schedule.WeekDay
+import kotlinx.coroutines.flow.Flow
+
+internal interface PreferenceStorage {
```
Comment (florina-muntenescu) [inline app/src/main/java/dev/sebastiano/bundel/preferences/PreferenceStorage.kt:7]: Checking the code, it looks like the return type is never used.
How about dropping it for all `set...` calls?

File: app/src/main/proto/dev/sebastiano/bundel/protos/settings.proto [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+option java_package = "dev.sebastiano.bundel.proto";
+option java_multiple_files = true;
+
+message BundelPreferences {
+
+    bool isCrashlyticsEnabled = 1;
+    bool isOnboardingSeen = 2;
+
+//    enum WeekDay {
```
Comment (florina-muntenescu) [inline app/src/main/proto/dev/sebastiano/bundel/protos/settings.proto:11]: remove this

**Review by florina-muntenescu — APPROVED** [review]

File: app/src/main/java/dev/sebastiano/bundel/preferences/PreferenceStorage.kt (hunk 2 above)
Comment (rock3r) [reply app/src/main/java/dev/sebastiano/bundel/preferences/PreferenceStorage.kt:7]: Yeah we could, let me do it

Comment (rock3r) [reply app/src/main/java/dev/sebastiano/bundel/preferences/PreferenceStorage.kt]: Done :) 20ddc58

File: app/src/main/proto/dev/sebastiano/bundel/protos/settings.proto (hunk 3 above)
Comment (rock3r) [reply app/src/main/proto/dev/sebastiano/bundel/protos/settings.proto]: Done 4b95e4f

**Pull request merged**